### ISSUE TITLE
(Angular 13) Formly Filter Checkboxes - does not appear checked for the first click and when refreshing the browser.

### ISSUE THREAD

**Issue by wonko-gsa** [GSA/sam-design-system#1064]
1. **Components Involved**
Formly Filter - multicheckboxes

2. **Expected/Actual Behavior**
**2-1. 
Expected Behavior**
i. Given that we log in and navigated to one of following workspace:
    ```
    Federal Entities
    Exclusions
    Non-Federal Entities
    BioPreferred
    SCR 
    ```
    ii. When we click a checkbox option in multi-checkbox filter (listed below) for the first time:
    ```
    BioPreferred workspace > Filter By box > Status 
    Federal Entities workspace > Filter By box > Entity Status
    Federal Entities workspace > Filter By box > Registration Status
    Non-Feneral Entities workspace > Filter By box > Status
    Non-Feneral Entities workspace > Filter By box > Address Update
    Exclusions workspace > Filter By box > Exclusion Status
    Exclusions workspace > Filter By box > Termination > Indefinite checkbox
    SCR reporting workspace > Filter By box > Status
    ```
    iii. Then we expect the checkbox appears checked and updates the model to filter the results and updates the URL parameters with the selected option.
    
   **Actual Behavior**
iv. The results is filtered with the checkbox selected, and the URL is updated, however the checkbox does not appear checked. The second attempt of clicking the same option makes it appearing as checked. <br/>
![1-multicheckbox-federal-entity-status-first-click-with-texts-pixelized](https://user-images.githubusercontent.com/117218338/199602967-204df902-e1ed-4328-a937-42b4c55d8ddf.png)

   **2-2.**
    **Expected Behavior**
    i. Given **2-1-i, 2-1-ii** above, 
    ii. When we refresh the page in the browser
    iii. Then we expect the URL remains the same, shows the same filtered results and "Filter by box fields" also show same values entered, or checked before refreshing. 

    **Actual Behavior**
    iv. The URL remains URL remains the same, shows the same filtered results and "Filter by box" fields show the same values entered. However the checkmarks selected before refreshing do not appear after refreshing. 
    
* Before refreshing while there are selected options:
![2-multicheckbox-federal-entity-status-before-refresh-pixelized](https://user-images.githubusercontent.com/117218338/199604555-bb4259f5-d3b9-45e2-97c0-ebeea1af03c2.png)

* After refreshing: 
![2-multicheckbox-federal-entity-status-after-refresh-pixelized](https://user-images.githubusercontent.com/117218338/199604579-db57bcd1-5d61-4277-94c1-eb9fcccc3bd4.png)

3. **Your Environment**
```
    "@angular/animations": "^13.3.11",
    "@angular/cdk": "^13.3.9",
    "@angular/common": "^13.3.11",
    "@angular/compiler": "^13.3.11",
    "@angular/core": "^13.3.11",
    "@angular/forms": "^13.3.11",
    "@angular/material": "^13.3.9",
    "@angular/platform-browser": "^13.3.11",
    "@angular/platform-browser-dynamic": "^13.3.11",
    "@angular/router": "^13.3.11",

    "@gsa-sam/components": "13.0.0-beta.3",
    "@gsa-sam/layout": "13.0.0-beta.1",
    "@gsa-sam/layouts": "13.0.0-beta.0",
    "@gsa-sam/ngx-uswds": "13.0.0-beta.1",
    "@gsa-sam/ngx-uswds-icons": "13.0.0-beta.2",
    "@gsa-sam/sam-formly": "13.0.0-beta.3",
    "@gsa-sam/sam-material-extensions": "13.0.0-beta.3",
    "@gsa-sam/sam-styles": "0.0.145",
    "@gsa-sam/shared": "13.0.0-beta.1",
```

4. **Ticket Number**
IAEMOD-4701 (Angular 13 Update)

5. **Formly config**
2-1, and 2-2 behaviors appear on multicheckboxes in following workspaces and here is a list of formly config:
a. BioPreferred workspace > Filter By box > Status 
https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/blob/integration-42.3/apps/entity-tier2/src/app/biopreferred-reporting-workspace/biopreferred-reporting-toolbar/biopreferred-reporting-filters/biopreferred-reporting-filters.component.ts#L293-L328 <br/>
b-1. Federal Entities workspace > Filter By box > Entity Status
https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/blob/integration-42.3/apps/entity-tier2/src/app/entity-workspace/federal-entities-workspace/federal-entity-toolbar/federal-entity-filters/federal-entity-filters.component.ts#L259-L318 <br/>
b-2. Federal Entities workspace > Filter By box > Registration Status
https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/blob/integration-42.3/apps/entity-tier2/src/app/entity-workspace/federal-entities-workspace/federal-entity-toolbar/federal-entity-filters/federal-entity-filters.component.ts#L351-L400 <br/>
c-1. Non-Feneral Entities workspace > Filter By box > Status
https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/blob/integration-42.3/apps/entity-tier2/src/app/entity-workspace/non-federal-entities-workspace/non-federal-entity-toolbar/non-federal-entity-filters/non-federal-entity-filters.component.ts#L297-L345 <br/>
c-2. Non-Feneral Entities workspace > Filter By box > Address Update
https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/blob/integration-42.3/apps/entity-tier2/src/app/entity-workspace/non-federal-entities-workspace/non-federal-entity-toolbar/non-federal-entity-filters/non-federal-entity-filters.component.ts#L248-L292 <br/>
d-1. Exclusions workspace > Filter By box > Exclusion Status
https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/blob/integration-42.3/apps/entity-tier2/src/app/exclusion-workspace/exclusion-toolbar/exclusion-filters/exclusion-filters.component.ts#L283-L318 <br/>
d-2. Exclusions workspace > Filter By box > Termination > Indefinite checkbox
https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/blob/integration-42.3/apps/entity-tier2/src/app/exclusion-workspace/exclusion-toolbar/exclusion-filters/exclusion-filters.component.ts#L363-L390 <br/>
e. SCR reporting workspace > Filter By box > Status
https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/blob/integration-42.3/apps/entity-tier2/src/app/scr-reporting-workspace/scr-reporting-toolbar/scr-reporting-filters/scr-reporting-filters.component.ts#L295-L332 <br/>

6. **Additional context**
This issue might/or might not be relevant to following issue.
This is a commit [[Code Link](https://github.helix.gsa.gov/GSA-IAE-APPS/sam-frontend-entity/commit/711c94dc0200232d922b61a5758861cf8f400921
)] to fix the following issue that was shared and discussed in IAE FWG space: 
**6-1.** In Federal Entities workspace, "Entity Status" in "Filter By" box is broken that it shows numbers: (1,2, 3, 4, 7, 12) for each label of checkbox instead of text labels: ('Work in Progress Registration', 'Submitted Registration', 'Active Registration', 'Inactive Registration', 'ID Assigned'). (attached a screenshot)<br/>
![1  federal-label-error-pixelized](https://user-images.githubusercontent.com/117218338/199602561-43814b60-85bb-4e74-be65-f92ef2bc12ae.png)
![2  exclusion-label-error-pixelized](https://user-images.githubusercontent.com/117218338/199602564-3b4eab28-8ef9-417f-a216-05e0e4f9e3d1.png)
**6-2.** In Exclusions workspace: 
  **a.** A label for a checkbox: "Indefinite" is showing as 'y' in "Filter By" box > "Classification" > "Termination Date". (attached a screenshot)
  **b.** The issue **6-2-a** above is resolved by replacing `value/label` to `key/value` as discussed which fixed issues of:
      - displaying label correctly in the UI and 
      - updating the correct parameter value (ex. 1, 2, y etc.) in the URL as we select the option in the UI.

**Comment by cwolf10** [comment]
An observation for when this issue is picked up. I am also observing this behavior in Horizontal Filters.

**Comment by yerramshilpa** [comment]
The issue has been resolved in @gsa-sam/sam-formly@13.0.2 version
